db/migrate/20190617191109_move_awx_credentials_to_authentications.rb

```diff
@@ -139,6 +139,10 @@ def self.decrypt(field_name, value, primary_key)
       return value unless value.include?("$encrypted$")
 
       new(secret_key, value, field_name, primary_key).decrypt
+    rescue Fernet256::InvalidToken
```

Comment:
So after thinking about this, I don't _**think**_ we need to handle the case where a `secret_key` doesn't exist in our database, since the `def up` should not even be trying to decrypt anything (even with an existing `awx` DB) since there is nothing in the `vmdb_production` to migrate.

But wanted to post this here since I ran into this as a "failing case" when trying to write the specs for this, and for a bit thought it might be relevant.

Reply:
An InvalidToken happens when the ansible key is invalid.

we have an invalid ansible key when we `fix_auth --invalid bogus` a database from a customer.

Reply:
@kbrock I think you are missing the point of the concern...

> An InvalidToken happens when the ansible key is invalid.

Wasn't concerned about this error that is already there, I am concerned about a different error that I was able to "reproduce" in the specs where it tries to do the look up in `AnsibleDecrypt.secret_key`, but doesn't find anything, so a `NoMethodError` is returned on `nil`.

Again, I don't think this scenario would be possible since I don't think you can have a `EmbeddedAnsible` authentication(s) record (which would be what allows it to get that far to hit `.secret_key` in the migration) without having secret key to decrypt with.  But I wouldn't mind a second opinion on that.